Syntactic cleanup

AlexKurek opened the pull request:
Replace tabs with spaces

AlexKurek committed:
Consolidate indents

Replace tabs with spaces

AlexKurek committed:
Fix indent also for AMD recipe

AlexKurek committed:
Fix cuda number

AlexKurek committed:
Fix cuda number in AMD

AlexKurek committed:
DPPP -> DP3

AlexKurek committed:
DPPP -> DP3

AlexKurek committed:
Add -fomit-frame-pointer -flto to DP3 and WSClean Intel

However, consider adding if DEBUG = FALSE here

AlexKurek committed:
which command was not avalilable before system installs

AlexKurek committed:
which command was not avalilable before system installs

tikk3r commented on singularity/Singularity.amd_aocl:
```diff
@@ -14,46 +14,46 @@ From: fedora:38
     export OMPI_ALLOW_RUN_AS_ROOT_CONFIRM=1
 
-    export HAS_CUDA={{ HAS_CUDA }}
+    export HAS_={{ HAS_ }}
```
Something went wrong here and the CUDA suffix disappeared.

AlexKurek committed:
fix HAS_CUDA Intel

AlexKurek committed:
fix HAS_CUDA AMD

AlexKurek committed:
Minor spacing and comments fixes

AlexKurek committed:
Make the version the same as in line 353 (Intel)

Line 353 Intel says:
pip install --no-deps --no-binary python-casacore "python-casacore==3.5.2"

tikk3r commented on singularity/Singularity.amd_aocl:
```diff
@@ -78,54 +78,54 @@ From: fedora:38
     fi
     export CPLUS_INCLUDE_PATH="/usr/include/python${PYTHON_VERSION}/:/opt/intel/oneapi/mkl/latest/include:/opt/hdf5/include:/opt/OpenBLAS/include:/usr/include/openmpi-x86_64:/usr/include/c++/9:$CPLUS_INCLUDE_PATH:/usr/include/python${PYTHON_VERSION}:$INSTALLDIR/casacore/include:/usr/include/boost:/usr/include/cfitsio:$INSTALLDIR/idg/include:$INSTALLDIR/EveryBeam/include:/usr/include/wcslib:/usr/i ...
-    export CPATH="/usr/include/python{$PYTHON_VERSION}:/opt/intel/oneapi/mkl/latest/include:/opt/hdf5/include:/opt/OpenBLAS/include:/usr/include/openmpi-x86_64/:/usr/local/cuda/include:/opt/intel/mkl/include:${INSTALLDIR}/casacore/include:$INSTALLDIR/idg/include:$INSTALLDIR/aoflagger/include:$INSTALLDIR/EveryBeam/include:/usr/include/wcslib:/usr/include/freetype2/freetype:/usr/include/freetype2/f ...
+    export CPATH="/usr/include/python{$PYTHON_VERSION}:/opt/intel/oneapi/mkl/latest/include:/opt/hdf5/include:/opt/OpenBLAS/include:/usr/include/openmpi-x86_64/:/usr/local//include:/opt/intel/mkl/include:${INSTALLDIR}/casacore/include:$INSTALLDIR/idg/include:$INSTALLDIR/aoflagger/include:$INSTALLDIR/EveryBeam/include:/usr/include/wcslib:/usr/include/freetype2/freetype:/usr/include/freetype2/freet ...
```
A rogue search and replace seems to have removed cuda here I assume?

AlexKurek replied:
I guess so, I didn't plan to change anything with CUDA.

AlexKurek committed:
Merge branch 'tikk3r:fedora-py3' into fedora-py3

AlexKurek committed:
Fedora 41 has Python 3.12

AlexKurek committed:
Fedora 41 has Python 3.12 AMD

AlexKurek committed:
Fix: Fedora 41 has Python 3.13 AMD

AlexKurek committed:
Fix: Fedora 41 has Python 3.13 Intel

AlexKurek committed:
Fix CUDA install Intel

AlexKurek committed:
Fix CUDA install AMD

AlexKurek committed:
Fix removed cuda export CPATH

tikk3r commented on singularity/Singularity.amd_aocl:
```diff
@@ -14,42 +14,42 @@ From: fedora:41
+    # General environment settings.
+    export J=`nproc`
+    export INSTALLDIR=/opt/lofar
+    export PYTHON_VERSION=3.13
```
I'm not sure which Fedora version the next containers will have, so let's not do version changes like this in this MR.

tikk3r commented on singularity/Singularity.intel_mkl:
```diff
@@ -81,56 +81,56 @@ From: fedora:41
     fi
     export CPLUS_INCLUDE_PATH="/usr/include/python${PYTHON_VERSION}/:/opt/intel/oneapi/mkl/latest/include:/opt/hdf5/include:/opt/OpenBLAS/include:/usr/include/openmpi-x86_64:/usr/include/c++/9:$CPLUS_INCLUDE_PATH:/usr/include/python${PYTHON_VERSION}:$INSTALLDIR/casacore/include:/usr/include/boost:/usr/include/cfitsio:$INSTALLDIR/idg/include:$INSTALLDIR/EveryBeam/include:/usr/include/wcslib:/usr/i ...
-    export CPATH="/usr/include/python{$PYTHON_VERSION}:/opt/intel/oneapi/mkl/latest/include:/opt/hdf5/include:/opt/OpenBLAS/include:/usr/include/openmpi-x86_64/:/usr/local/cuda/include:/opt/intel/mkl/include:${INSTALLDIR}/casacore/include:$INSTALLDIR/idg/include:$INSTALLDIR/aoflagger/include:$INSTALLDIR/EveryBeam/include:/usr/include/wcslib:/usr/include/freetype2/freetype:/usr/include/freetype2/f ...
+    export CPATH="/usr/include/python{$PYTHON_VERSION}:/opt/intel/oneapi/mkl/latest/include:/opt/hdf5/include:/opt/OpenBLAS/include:/usr/include/openmpi-x86_64/:/usr/local//include:/opt/intel/mkl/include:${INSTALLDIR}/casacore/include:$INSTALLDIR/idg/include:$INSTALLDIR/aoflagger/include:$INSTALLDIR/EveryBeam/include:/usr/include/wcslib:/usr/include/freetype2/freetype:/usr/include/freetype2/freet ...
```
cuda got lost here as well.

tikk3r commented on singularity/Singularity.intel_mkl:
```diff
@@ -451,7 +451,7 @@ EOF
         cmake3 -DCMAKE_INSTALL_PREFIX:PATH=$INSTALLDIR/DPPP -DIDGAPI_LIBRARIES=$INSTALLDIR/idg/lib/libidg-api.so -DIDGAPI_INCLUDE_DIRS=$INSTALLDIR/idg/include -DAOFLAGGER_INCLUDE_DIR=$INSTALLDIR/aoflagger/include -DAOFLAGGER_LIB=$INSTALLDIR/aoflagger/lib/libaoflagger.so -DPORTABLE=True -DLIBDIRAC_PREFIX=$INSTALLDIR/sagecal/ -DBUILD_BENCHMARKS=OFF ../src
     else
-        cmake3 -DCMAKE_INSTALL_PREFIX:PATH=$INSTALLDIR/DPPP -DIDGAPI_LIBRARIES=$INSTALLDIR/idg/lib/libidg-api.so -DIDGAPI_INCLUDE_DIRS=$INSTALLDIR/idg/include -DAOFLAGGER_INCLUDE_DIR=$INSTALLDIR/aoflagger/include -DAOFLAGGER_LIB=$INSTALLDIR/aoflagger/lib/libaoflagger.so -DTARGET_CPU=${MARCH} -DLIBDIRAC_PREFIX=$INSTALLDIR/sagecal/ -DBUILD_BENCHMARKS=OFF ../src
+        cmake3 -DCMAKE_INSTALL_PREFIX:PATH=$INSTALLDIR/DPPP -DIDGAPI_LIBRARIES=$INSTALLDIR/idg/lib/libidg-api.so -DIDGAPI_INCLUDE_DIRS=$INSTALLDIR/idg/include -DAOFLAGGER_INCLUDE_DIR=$INSTALLDIR/aoflagger/include -DAOFLAGGER_LIB=$INSTALLDIR/aoflagger/lib/libaoflagger.so -DTARGET_CPU=${MARCH} -DCMAKE_CXX_FLAGS='-fomit-frame-pointer -flto' -DCMAKE_C_FLAGS='-fomit-frame-pointer -flto' -DLIBDIRAC_PRE ...
```
I'm not against more optimisation, but I think that maybe fits better in a separate MR at some point. Two questions I have here are:

1. Are the global CXX_FLAGS and C_FLAGS still used when this is done this way?
2. What kind of performance gain do we get from this?

AlexKurek replied:
1. Good point, needs testing
2. I have no idea, but this allows (some kind of) vectorization, so theoretically it could be a big gain.

AlexKurek committed:
Speedup SAGECal

AlexKurek committed:
Speedup SAGECal AMD

AlexKurek committed:
Cleanup LTA install AMD

AlexKurek committed:
Cleanup LTA install Intel

AlexKurek committed:
Fix datashape install for Python 3.12+ AMD

AlexKurek committed:
Fix datashape install for Python 3.12+ Intel

AlexKurek committed:
Remove Oracle install file AMD

AlexKurek committed:
Remove Oracle install file Intel

AlexKurek committed:
Remove AOCTL tgz

AlexKurek committed:
Fix indents

AlexKurek committed:
Fix indents Intel

AlexKurek committed:
Remove EveryBeam src Intel

AlexKurek committed:
Remove EveryBeam src AMD

AlexKurek committed:
LOSOTO3 ---> LOSOTO

AlexKurek committed:
Update AOCL to 5.0.0

AlexKurek committed:
Typo

AlexKurek committed:
requirements3.txt -> requirements.txt

AlexKurek committed:
Remove CASAREST artifacts

AlexKurek committed:
Remove obsolete DUSE_FFTW3

I verified that if finds the libraries without DUSE_FFTW3.

AlexKurek committed:
DPPP -> DP3

AlexKurek committed:
Consolidate cmake naming

AlexKurek committed:
Comment Montage artifacts

AlexKurek committed:
Remove duplicated Multinest PATH

tikk3r commented:
Let's not put optimisations in this MR. They benefit from some more quantification and a separate record I think.

tikk3r commented on singularity/Singularity.amd_aocl:
```diff
@@ -83,65 +81,67 @@ From: fedora:41
-    wget https://download.amd.com/developer/eula/aocl/aocl-4-0/aocl-linux-aocc-4.0.tar.gz
-    tar xf aocl-linux-aocc-4.0.tar.gz
-    cd aocl-linux-aocc-4.0
+    wget https://download.amd.com/developer/eula/aocl/aocl-5-0/aocl-linux-aocc-5.0.0.tar.gz
```
Let's not update this version in this MR.

tikk3r replied:
(It's ok to open a separate MR for updating AOCL  though, I'd just like to track the performance change from it in a more isolated fashion).

AlexKurek replied:
Ok, I have opened it.

tikk3r commented on singularity/Singularity.amd_aocl:
```diff
@@ -239,16 +239,16 @@ From: fedora:41
+        # Install CUDA 12.6
         #
-        sudo dnf -y config-manager addrepo --from-repofile=http://developer.download.nvidia.com/compute/cuda/repos/fedora41/x86_64/cuda-fedora41.repo
+        sudo dnf -y config-manager addrepo --from-repofile=https://developer.download.nvidia.com/compute/cuda/repos/fedora39/x86_64/cuda-fedora39.repo
```
I'm experimenting with which Fedora version to use, to same here; let's not change versions.

tikk3r commented on singularity/Singularity.amd_aocl:
```diff
@@ -239,16 +239,16 @@ From: fedora:41
         sudo dnf -y clean all
         sudo dnf -y upgrade
-        sudo dnf -y install cuda-12-3
+        sudo dnf -y install cuda-12-6
```
See above.

tikk3r commented on singularity/Singularity.amd_aocl:
```diff
@@ -285,7 +285,19 @@ From: fedora:41
     pip install --no-binary h5py h5py
-    pip install -r $INSTALLDIR/requirements3.txt
+
+    # temporary fix for https://github.com/tikk3r/flocs/issues/100#issuecomment-2543192437
```
I'm already looking into this and I think installing directly from github works, so I think this isn't needed.

tikk3r replied:
Oh apparently not, it looked like it did but trying again it doesn't.

AlexKurek replied:
There are two ways of fixing it now. The one I put above (using `sed`) and:
https://github.com/ratt-ru/shadeMS/issues/124#issuecomment-2544217464
Both work for me.

tikk3r commented on singularity/Singularity.amd_aocl:
```diff
@@ -339,91 +352,90 @@ From: fedora:41
+    cd $INSTALLDIR/sagecal
+    git clone https://github.com/nlesc-dirac/sagecal.git src
+    cd build
+    cmake -DOpenBLAS_LIB=/opt/OpenBLAS/lib64/libopenblas.so -DCMAKE_INSTALL_PREFIX=$INSTALLDIR/sagecal -DLIB_ONLY=True -Wno-dev -DCMAKE_CXX_FLAGS='-g -O3 -fopenmp -ffast-math -lmvec -lm' -DCMAKE_C_FLAGS='-g -O3 -fopenmp -ffast-math -lmvec -lm' ../src
```
Same comment about leaving optimisations for a different time (especially `-ffast-math` may not be safe so should be tested beforehand).

AlexKurek committed:
Undo a few things (resolve rewiev)

AlexKurek committed:
Merge branch 'tikk3r:fedora-py3' into fedora-py3
